rc/lsp.kak

```diff
@@ -452,19 +452,19 @@ Jump to the next or previous diagnostic error" %{
         current=""
         prev=""
         selection=""
-        for e in $@; do
-            if [[ $e == *"$errorCompare"* ]]; then
+        for e in "$@"; do
+            if [ "${e#*$errorCompare}" != "$e" ]; then # e contains errorCompare
                 e=${e%,*}
                 line=${e%.*}
                 column=${e#*.}
-                if [[ $line -eq $kak_cursor_line && $column -eq $kak_cursor_column ]]; then
+                if [ "$line" -eq "$kak_cursor_line" ] && [ "$column" -eq "$kak_cursor_column" ]; then
                     continue #do not return the current location
                 fi
                 current="$line $column"
                 if [ -z "$first" ]; then
                     first="$current"
                 fi
-                if [[ $line -gt $kak_cursor_line || (($line -eq $kak_cursor_line) && ($column -gt $kak_cursor_column)) ]]; then
+                if [ "$line" -gt "$kak_cursor_line" ] || { [ "$line" -eq "$kak_cursor_line" ] && [ "$column" -gt "$kak_cursor_column" ]; }; then
```

Comment:
You can use parentheses instead of spawning a subshell.  From the [spec](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html):

> The following form can be used instead:
>
>     ( test expr1 && test expr2 ) || test expr3

Reply:
As far as I can tell `()` spawns a subshell while `{}` does not https://askubuntu.com/questions/606378/when-to-use-vs-in-bash/606386

Reply:
According to [spec](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html) 
```
test "$1" -a "$2"

should be written as:

test "$1" && test "$2"
```

Reply:
> As far as I can tell `()` spawns a subshell while `{}` does not https://askubuntu.com/questions/606378/when-to-use-vs-in-bash/606386

Oh yeah, you're right here.